src/app/pages/tx-summary/tx-summary-details.tsx

```diff
@@ -0,0 +1,31 @@
+import { Box, Flex, Stack, Text } from '@stacks/ui';
+
+export type TxDetails = { title: string; value: string } | 'line';
+interface TxSummaryDetailsProps {
+  details: TxDetails[];
+}
+
+export function TxSummaryDetails({ details }: TxSummaryDetailsProps) {
+  return (
+    <Stack width="100%" mb="44px">
+      {details.map(item => {
+        if (item === 'line') {
+          return (
+            <Box py="16px">
+              <hr />
+            </Box>
+          );
+        }
+
+        return (
+          <Flex justifyContent="space-between" fontSize="14px">
+            <Text color="#74777D">{item.title}</Text>
+            <Text color="#242629" fontWeight="500">
+              {item.value}
+            </Text>
+          </Flex>
+        );
+      })}
+    </Stack>
+  );
+}
```

Comment:
Recommend composing this with components, rather than creating an array like this (the configuration approach). 

I've seen this done before, and it makes it difficult to change, when the design changes slightly. What happens then is that the configuration grows and grows to handle all new circumstances and you have a huge component.

Composing these together is slightly more explicit, but makes changes trivial.

```ts
function TxSummaryDetail(props) {
  return (
    <Flex justifyContent="space-between" fontSize="14px">
      <Text color="#74777D">{item.title}</Text>
      <Text color="#242629" fontWeight="500">
        {item.value}
      </Text>
    </Flex>
  )
}


function Line() {
  return (
    <Box py="16px">
      <hr />
    </Box>
  );
}

// used like this
<TxSummaryDetails>
  <TxSummaryDeatil someProp={} />
  <Line />
  <TxSummaryDeatil someProp={} />
</TxSummaryDetails>
```